src/sentry/models/release.py

```diff
@@ -928,3 +929,27 @@ def safe_delete(self):
             releasefile.file.delete()
             releasefile.delete()
         self.delete()
+
+    def count_artifacts(self):
+        """Sum the artifact_counts of all release files.
+
+        An artifact count of NULL is interpreted as 1.
+        """
+        counts = get_artifact_counts([self.id])
+        return counts.get(self.id, 0)
+
+
+def get_artifact_counts(release_ids: List[int]) -> Mapping[int, int]:
+    """Get artifact count grouped by IDs"""
+    query = """
+    SELECT
+        release_id AS id, sum(COALESCE(artifact_count, 1)) AS count
+    FROM
+        sentry_releasefile
+    WHERE
+        release_id = ANY(%s)
+    GROUP BY
+        release_id;
+    """
+    qs = Release.objects.raw(query, params=[release_ids])
```

Comment:
Let's not use raw SQL if not strictly necessary. This can be replicated in Django ORM with
```
In [24]: q = ReleaseFile.objects.filter(release_id__in=[1]).annotate(count=Sum(Func(F("artifact_count"), 1, function="COALESCE"))).values_list("release_id", "count")

In [25]: q.query.group_by = ["release_id"]

In [26]: print(sqlparse.format(str(q.query), reindent=True))
SELECT "sentry_releasefile"."release_id",
       SUM(COALESCE("sentry_releasefile"."artifact_count", 1)) AS "count"
FROM "sentry_releasefile"
WHERE "sentry_releasefile"."release_id" IN (1)
GROUP BY "sentry_releasefile"."release_id"
```

But the real problem I see here is the mismatch of the table we select from and the model we use. Django uses the model and its table name to decide which database connection to use and with the plan to move `ReleaseFile` to a separate database this will break.

Reply:
Thanks, change pushed now.